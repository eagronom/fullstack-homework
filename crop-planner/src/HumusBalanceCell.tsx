import { PureComponent } from 'react'

type Props = {
  value: number
}

type State = {
  previousHumusBalance: number,
  currentHumusBalance: number
  fieldQuality: string
}

export default class HumusBalanceCell extends PureComponent<Props, State> {
  constructor(props: Props) {
    super(props)

    this.state = {
      previousHumusBalance: this.props.value,
      currentHumusBalance: this.props.value,
      fieldQuality: ''
    }
  }

  static getDerivedStateFromProps(props: Props, state: State) {    
    state.previousHumusBalance = state.currentHumusBalance
    state.currentHumusBalance = props.value
    
    let fieldQuality = ''
    if (state.previousHumusBalance !== state.currentHumusBalance)
      fieldQuality = (state.previousHumusBalance > state.currentHumusBalance) ? 'decreased' : 'increased' 

    state.fieldQuality = fieldQuality

    return state;
  }

  classList = () => {
    const defaultClass = 'table__cell table__cell--right'
    if (this.state.fieldQuality === '') return defaultClass

    return `${defaultClass } ${this.state.fieldQuality}-field-quality`
  }

  render = () => 
    <div className={this.classList()}>
      {this.state.currentHumusBalance.toFixed(2)}
    </div>
}