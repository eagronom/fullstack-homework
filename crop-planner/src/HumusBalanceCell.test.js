import { shallow } from 'enzyme'

import HumusBalanceCell from './HumusBalanceCell'

describe('<HumusBalanceCell />', () => {
  it('renders', () => {
    expect(shallow(
      <HumusBalanceCell value={10} />
    )).toMatchSnapshot()
  })
})
