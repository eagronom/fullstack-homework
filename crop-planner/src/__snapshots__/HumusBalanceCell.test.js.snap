// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HumusBalanceCell /> renders 1`] = `
<div
  className="table__cell table__cell--right"
>
  10.00
</div>
`;
